nautobot/extras/models/jobs.py

```diff
@@ -669,7 +678,14 @@ def set_status(self, status):
         """
         self.status = status
         if status in JobResultStatusChoices.TERMINAL_STATE_CHOICES:
+            # Only add metrics if we have a related job model. If we are moving to a terminal state we should always
+            # have a related job model, so this shouldn't be too tight of a restriction.
             self.completed = timezone.now()
+            if self.job_model:
+                duration = self.completed - self.created
+                JOB_RESULT_METRIC.labels(self.job_model.module_name, self.job_model.job_class_name, status).observe(
```

Comment:
Would it be preferable to use `grouping` and `name` (human-readable, configurable) instead of `module_name` and `job_class_name` (hard-coded)? Not sure, so I thought I'd ask.

Reply:
I will gather some opinions about this and loop back around here.

Reply:
Changed.